### PR TITLE
[MERGE] various: improve html fields integration

### PR DESCRIPTION
**Pull request by dht-odoo**
PURPOSE

Fix issues generated due to conversion of text field into Html field.

SPECIFICATIONS

Improve integration of HTML fields as well as "empty html" management.
See various commits for more details.

Followup of odoo/odoo@cb4f597 and odoo/enterprise@971f3ed

Task-2558824

### CODE REVIEW

File: addons/account/views/account_move_views.xml [hunk 1]
```diff
@@ -1019,25 +1019,26 @@
...
-                                    <field name="invoice_payments_widget" colspan="2" nolabel="1" widget="payment"/>
-                                    <field name="amount_residual" class="oe_subtotal_footer_separator" attrs="{'invisible': [('state', '=', 'draft')]}"/>
-                                </group>
-                                <field name="invoice_outstanding_credits_debits_widget"
-                                    class="oe_invoice_outstanding_credits_debits"
-                                    colspan="2" nolabel="1" widget="payment"
-                                    attrs="{'invisible': ['|', ('state', '!=', 'posted'), ('move_type', 'in', ('out_receipt', 'in_receipt'))]}"/>
-                                <group>
-                                    <field name="narration" placeholder="Terms and Conditions" class="oe_inline" nolabel="1"/>
+                                <group col="6">
```
Comment (tde-banana-odoo) [inline addons/account/views/account_move_views.xml:1022]: Meh, standard odoo has 4 col as far as I know the internals of qweb client (might have changed)... this whole diff seems quite strange, can't you find a more elegant way to fix that ?

File: addons/repair/models/repair.py [hunk 2]
```diff
@@ -358,10 +360,10 @@ def _create_invoices(self, group=False):
                 invoice_vals = current_invoices_list[0]
                 invoice_vals['invoice_origin'] += ', ' + repair.name
                 invoice_vals['repair_ids'].append((4, repair.id))
-                if not invoice_vals['narration']:
+                if  is_html_empty(invoice_vals['narration']):
                     invoice_vals['narration'] = narration
                 else:
-                    invoice_vals['narration'] += '<br/>' + narration
+                    invoice_vals['narration'] += Markup('<br/>') + narration
```
Comment (tde-banana-odoo) [inline addons/repair/models/repair.py:366]: Could be great to check that narration is not void either (to do also at line 347 when adding narration in the if).

**Review by tde-banana-odoo — COMMENTED** [review]
Review

File: addons/repair/models/repair.py (hunk 2 above)
Comment (dht-odoo) [reply addons/repair/models/repair.py]: Changes applied.

File: addons/account/views/account_move_views.xml (hunk 1 above)
Comment (dht-odoo) [reply addons/account/views/account_move_views.xml]: Changes applied.

**Comment by tde-banana-odoo** [comment]
@robodoo override=ci/security

Reason: using Markup as a fix to concatenate Html content

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**